test/upgrade/installation/serverless.go

```diff
@@ -97,10 +98,29 @@ func DowngradeServerless(ctx *test.Context) error {
 		return err
 	}
 
+	// Delete olm pods to avoid cache issues (https://access.redhat.com/solutions/6991414)
+	for _, labelValue := range []string{"olm-operator", "catalog-operator"} {
+		pods, err := ctx.Clients.Kube.CoreV1().Pods("openshift-operator-lifecycle-manager").List(context.Background(), metav1.ListOptions{
+			LabelSelector: metav1.FormatLabelSelector(&metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": labelValue},
+			}),
+		})
+		if err != nil {
+			return err
+		}
+		for _, p := range pods.Items {
+			if err := ctx.Clients.Kube.CoreV1().Pods("openshift-operator-lifecycle-manager").Delete(context.Background(), p.Name, metav1.DeleteOptions{}); err != nil {
+				return err
+			}
+		}
+	}
+
+	time.Sleep(1 * time.Minute)
+
 	// If we are on OCP 4.8 we need to apply the workaround in https://access.redhat.com/solutions/6992396.
 	// Currently, we only test in 4.8 (1.21) and 4.11+ (1.24+). Latest versions (eg. 4.11+) have a fix for this so no need to patch the crds,
 	// but we do it anyway for supported versions up to 4.10.
-	if err := common.CheckMinimumKubeVersion(ctx.Clients.Kube.Discovery(), "1.23.0"); err != nil {
+	if err := common.CheckMinimumKubeVersion(ctx.Clients.Kube.Discovery(), "1.24.0"); err != nil {
```

Comment:
CheckMinimumKubeVersion was not returning an error although we were on 1.23.x as 1.23.x was greater than 1.23.0.

Reply:
Since the root cause is this line, it seems "Delete olm pods" change would not be necessary? If so, would you like to delete the change around L#101 to verify it is not related?

Reply:
Never mind. I found you will test it on the Slack.

Reply:
This removes logs like:
```
$ oc -n openshift-operator-lifecycle-manager logs catalog-operator-f7f79797c-rlnvr | grep ResolutionFailed | more
I0313 10:45:04.510671       1 event.go:282] Event(v1.ObjectReference{Kind:"Namespace", Namespace:"",
 Name:"openshift-serverless", UID:"519f8611-b8dd-418c-a396-afe06208d85d", APIVersion:"v1", ResourceV
ersion:"48463", FieldPath:""}): type: 'Warning' reason: 'ResolutionFailed' constraints not satisfiab
le: subscription serverless-operator requires @existing/openshift-serverless//serverless-operator.v1
.28.0, subscription serverless-operator exists, @existing/openshift-serverless//serverless-operator.
v1.28.0 and @existing/openshift-serverless//serverless-operator.v1.27.0 originate from package serve
rless-operator, clusterserviceversion serverless-operator.v1.27.0 exists and is not referenced by a 
subscription
```
but I will check if it actually fails the tests shortly.